File: 3068_Find_the_Maximum_Sum_of_Node_Values.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @param {number[][]} edges
 * @return {number}
 */
var maximumValueSum = function(nums, k, edges) {
        let maxSum = nums.reduce((sum, num) => sum + Math.max(num, num ^ k), 0);
        let changedCount = nums.reduce((count, num) => count + ((num ^ k) > num ? 1 : 0), 0);

        if (changedCount % 2 === 0) {
            return maxSum;
        } else {
            let minChangeDiff = Math.min(...nums.map(num => Math.abs(num - (num ^ k))));
            return maxSum - minChangeDiff;
        }

};

File: 3068_Find_the_Maximum_Sum_of_Node_Values.rs
impl Solution {
    pub fn maximum_value_sum(nums: Vec<i32>, k: i32, edges: Vec<Vec<i32>>) -> i64 {
        let max_sum: i64 = nums.iter().map(|&num| num.max(num ^ k) as i64).sum();
        let changed_count: i64 = nums.iter().map(|&num| ((num ^ k) > num) as i64).sum();

        if changed_count % 2 == 0 {
            return max_sum;
        } else {
            let min_change_diff = nums.iter()
                .map(|&num| (num - (num ^ k)).abs() as i64)
                .min()
                .unwrap_or(0);

            return max_sum - min_change_diff;
        }

    }
}
